linkedin-clone/src/components/Rightside.js

```diff
@@ -1,9 +1,207 @@
+    skillsMatchScore: skillsMatchScore(currentUser.skills, u.skills),
+    languagesMatchScore: languagesMatchScore(currentUser.languages, u.languages), 
+  }))
+    .filter((u) => !Array.isArray(currentUser.connections) || !currentUser.connections.some((c) => c.id === u.id))
+    .sort((a, b) => {
+      // Sort by common connections (descending)
+      const commonConnectionsDiff = b.commonConnections - a.commonConnections;
+      if (commonConnectionsDiff !== 0) {
+        return commonConnectionsDiff;
+      }
+    
+      // Sort by common industry (if both have the same number of common connections)
+      if (currentUser.searchingPreferences && currentUser.searchingPreferences.industry) {
+        const currentUserIndustry = currentUser.searchingPreferences.industry;
+        const aMatch = a.industry === currentUserIndustry;
+        const bMatch = b.industry === currentUserIndustry;
+    
+        if (aMatch && !bMatch) {
+          return -1;
+        }
+        if (!aMatch && bMatch) {
+          return 1;
+        }
+      }
+    
+      // Sort by common school and course title (as the third and fifth filter)
+      const coursesMatchDiff = b.coursesMatchScore - a.coursesMatchScore;
+      if (coursesMatchDiff !== 0) {
+        return coursesMatchDiff;
+      }
+    
+      // Sort by common skills (as the fifth filter)
+      const skillsMatchDiff = b.skillsMatchScore - a.skillsMatchScore;
+      if (skillsMatchDiff !== 0) {
+        return skillsMatchDiff;
+      }
+  
+       // Sort by common languages (as the sixth filter)
+       const languagesMatchDiff = b.languagesMatchScore - a.languagesMatchScore;
+       if (languagesMatchDiff !== 0) {
+         return languagesMatchDiff;
+       }
+  
+        // Sort by experience level (as the seventh filter)
+        return b.experienceLevelMatch - a.experienceLevelMatch;
+    });
+  
+    // Adjust the limit to show more or fewer suggested users
+    setSuggestedUsers(commonUsers.slice(0, limit));
+    setLoading(false);
+  };
+
+
+
+  // useEffect hook to fetch suggested users and handle window resize
+  useEffect(() => {
+    const unsubscribe = auth.onAuthStateChanged((user) => {
+      if (user) {
+        const userId = user.uid;
+        
+        fetchSuggestedUsers(userId, displayedUsers);
+      } else {
+        setLoading(false);
+      }
+    });
+
+    const handleResize = () => {
+      if (window.innerWidth < 1100 && window.innerWidth >= 900) {
+        setDisplayedUsers(20);
+      } else if(window.innerWidth < 900){
+        setDisplayedUsers(10);
+      }
+      else setDisplayedUsers(30);
+      if (auth.currentUser) {
+        fetchSuggestedUsers(auth.currentUser.uid, displayedUsers);
+      }
+    };
+    window.addEventListener('resize', handleResize);
+    handleResize();
+
+    
+
+
+    return () => {
+      unsubscribe();
+      window.removeEventListener('resize', handleResize);
+    }
+
+  }, [displayedUsers]);
```

Comment:
Make sure this does not cause infinite loops. You can remove the parameter if not needed here.

Reply:
it's needed to display the right amount of users depending on the width of the window

Reply:
Can you console log in your useEffect to see if the function repeats? Let me know

Reply:
problem resolved